src/event_loop.rs

```diff
@@ -165,24 +168,33 @@ impl<Io> EventLoop<Io>
 
     // Drain the channel
     //
-    // Returns true if items were received
-    fn drain_recv_channel(&self, state: &mut State) -> bool {
+    // Returns `Ok` if the `EventLoop` should continue running.
+    // `Ok(true)`is returned if items were received
+    //
+    // An `Err` indicates that the event loop should shut down
+    fn drain_recv_channel(&self, state: &mut State) -> Result<bool, ()> {
```

Comment:
Feels a bit inelegant to use a result with void error to represent this three-state logic. Personally I think either make a descriptive error type, or perhaps even better, use `Option<bool>` where `None` indicate the channel will have no more input.